### PR TITLE
Clears old entries from the table

### LINKED ISSUE #14310: Payments show mins time not met message even though ledger table contains more than 30mins of usage
### Description
Payments show mins time not met message even though table contains more than 30mins of usage

### Steps to Reproduce
  1. Enable payments and set Budget to 100BAT
  2. View a variety of pages(Twitch/YT/Browsing) to add to ledger table
  3. Ensure its more than 30 mins
  4. Change reconcileStamp to older date to trigger payment 
  5. Relaunch browser shows min time not met message even ledger table shows more than 30 mins of usage 

**Actual result:**
image

**Expected result:**
Should not show message if min time criteria is met

**Reproduces how often:**
Unsure. @davidtemkin had seen this on his profile as well. 

### Brave Version

**about:brave info:**

Brave | 0.22.725
-- | --
V8 | 6.6.346.32
rev | e6dd17f
Muon | 6.0.12
OS Release | 10.0.17134
Update Channel | Release
OS Architecture | x64
OS Platform | Microsoft Windows
Node.js | 7.9.0
Brave Sync | v1.4.2
libchromiumcontent | 66.0.3359.170

**Reproducible on current live release:**
Maybe

### Additional Information
cc: @jasonrsadler @ryanml @NejcZdovc

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #14310

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
**you need to have a profile stat is older then one month and was not used (with payments turned on and containing some publishers). If you don't have it contact me and I will send you the profile**

(DO THIS ON OTHER BRANCH)
- run old profile (should be older then 1month, payments enabled)
- pin some sites and visit some new sites that are not on the table

(ON THIS BRANCH)
- start profile make sure that timestamp is pushed now + 30 days
- make sure that table is the same as it was on a previous branch
- close browser
- change reconcileStamp in the past
- start browser again
- make sure that only pinned sites and newly visited sites are on the table (if this total is not more then 30min, fuzzing should happen)

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by NejcZdovc** [comment]
blocked on https://github.com/brave-intl/bat-publisher/pull/34 and https://github.com/brave-intl/bat-client/pull/84

File: test/unit/app/browser/reducers/ledgerReducerTest.js [hunk 1]
```diff
@@ -870,19 +871,29 @@ describe('ledgerReducer unit tests', function () {
   })
 
   describe('APP_ON_LEDGER_FUZZING', function () {
```
Comment (ryanml) [inline test/unit/app/browser/reducers/ledgerReducerTest.js]: (nit) could you add a case for `newStamp: 0`? Then coverage should be at 100% :D

Comment (NejcZdovc) [reply test/unit/app/browser/reducers/ledgerReducerTest.js]: test added

**Comment by NejcZdovc** [comment]
master 9fbc862
0.23 68388a2cc
0.22 d1fb876ab

**Pull request merged**